src/zenml/enums.py

```diff
@@ -103,6 +103,11 @@ class SecretsStoreType(StrEnum):
     NONE = "none"  # indicates that the secrets store is disabled
```

Comment:
Can one actually do this? I thought we always fall back to SQL/Rest depending on the configuration of the active zen store? In `BaseSecretsStore.get_store_class`, the `NONE` case isn't covered and it instead raises a TypeError

Reply:
Yes, this is possible, but you have to explicitly set the secrets store type to NONE in the configuration. Leaving the configuration empty defaults to whatever is used for the Zen store. It never gets as far as `get_store_class` when this is set to NONE because of this code in the `BaseZenStore.create_store` initialization method:

```        # Initialize the secrets store
        if (
            secrets_store_config
            and secrets_store_config.type != SecretsStoreType.NONE
        ):
            ...
```